locator/guice/src/main/java/org/eclipse/kapua/locator/guice/GuiceLocatorImpl.java

```diff
@@ -129,230 +133,64 @@ private void init(String locatorConfigName) throws Exception {
-            configurationPrinter.addSimpleParameter("None");
-        }
-        configurationPrinter.closeSection();
+        configurationPrinter.logSections("Loaded Kapua Modules", instantiatedKapuaModules, Comparator.comparing(a -> a.getClass().getName()));
+        configurationPrinter.logSections("Excluded Kapua Modules", includedAndExcludedModules.get(false), Comparator.comparing(a -> a.getClass().getName()));
 
         // Print it!
         configurationPrinter.printLog();
-    }
-
-    /**
-     * Prints the configuration of the loaded {@link KapuaModule}s.
-     *
-     * @param resourceNameURL       The {@link KapuaLocator} configuration resource name.
-     * @param locatorConfig         The loaded {@link LocatorConfig}.
-     * @param loadedXmlSerializable The laaded {@link KapuaModule}s
-     * @since 2.0.0
-     */
-    private void printLoadedXmlSerializableConfiguration(@NotNull URL resourceNameURL, @NotNull LocatorConfig locatorConfig, @NotNull List<Class<?>> loadedXmlSerializable, @NotNull List<Class<?>> excludedXmlSerializable) {
-        ConfigurationPrinter configurationPrinter =
-                ConfigurationPrinter
-                        .create()
-                        .withLogger(LOG)
-                        .withLogLevel(ConfigurationPrinter.LogLevel.INFO)
-                        .withTitle("Kapua XmlSerializable Configuration")
-                        .addParameter("Resource Name", resourceNameURL.getPath());
 
-        // Packages
-        addIncludedExcludedPackageConfig(configurationPrinter, locatorConfig);
-
-        // Loaded modules
-        if (LOG.isDebugEnabled()) {
-            // Printing like this is highly verbose
-            configurationPrinter.openSection("Loaded XmlSerializable Classes");
-            if (!loadedXmlSerializable.isEmpty()) {
-                for (Class<?> xmlSerializableClass : sortedClass(loadedXmlSerializable)) {
-                    configurationPrinter.addSimpleParameter(xmlSerializableClass.getName());
-                }
-            } else {
-                configurationPrinter.addSimpleParameter("None");
-            }
-            configurationPrinter.closeSection();
-        } else {
-            configurationPrinter.addParameter("Loaded XmlSerializable Classes", loadedXmlSerializable.size());
-        }
-
-        // Loaded modules
-        if (LOG.isDebugEnabled()) {
-            // Printing like this is highly verbose
-            configurationPrinter.openSection("Excluded XmlSerializable Classes");
-            if (!excludedXmlSerializable.isEmpty()) {
-                for (Class<?> xmlSerializableClass : sortedClass(excludedXmlSerializable)) {
-                    configurationPrinter.addSimpleParameter(xmlSerializableClass.getName());
-                }
-            } else {
-                configurationPrinter.addSimpleParameter("None");
+        //Add JAXB custom module to the lot
+        instantiatedKapuaModules.add(new AbstractKapuaModule() {
```

Comment:
An on-the-fly module is needed here to inizialize the XmlSerializableClassesProvider with the same included and excluded packages as the kapua locator itself